test/bulwark/core_test.clj

```diff
@@ -212,3 +213,32 @@
       (is (re-matches #"goo timed-out.*" (-> exception
                                              (.getCause)
                                              (.getMessage)))))))
+
+(deftest test-fallback-function
+  (testing "should not clear MDC if running on same thread"
+    (MDC/clear)
+    (MDC/setContextMap {"a" "1"})
```

Comment:
We can extract these out into fixtures. https://clojuredocs.org/clojure.test/use-fixtures